### PR TITLE
Fix (ci): Fix `REGISTRY_USER` and `REGISTRY_PASSWORD` to be optional when `NO_PUSH=true`

### PR DESCRIPTION
**Pull request by leojonathanoh**

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -98,6 +92,17 @@ else
     export STEAM_PASSWORD=${STEAM_PASSWORD:-}
 fi
 
+# Process user variables
+REGISTRY_GOLDSOURCE=${REGISTRY_GOLDSOURCE:?err}
+REGISTRY_SOURCE=${REGISTRY_SOURCE:?err}
+if [ ! "$NO_TEST" = 'true' ]; then
```
Comment (joeltimothyoh) [inline build.sh:98]: `NO_TEST` or `NO_PUSH`?

**Review by joeltimothyoh — CHANGES_REQUESTED** [review]

File: build.sh (hunk 1 above)
Comment (leojonathanoh) [reply build.sh:98]: let me fix

File: build.sh [hunk 2]
```diff
@@ -98,6 +92,17 @@ else
...
+REGISTRY_GOLDSOURCE=${REGISTRY_GOLDSOURCE:?err}
+REGISTRY_SOURCE=${REGISTRY_SOURCE:?err}
+if [ ! "$NO_PUSH" = 'true' ]; then
+    REGISTRY_USER=${REGISTRY_USER:-}
+    REGISTRY_PASSWORD=${REGISTRY_PASSWORD:-}
+else
+    REGISTRY_USER=${REGISTRY_USER:?err}
+    REGISTRY_PASSWORD=${REGISTRY_PASSWORD:?err}
+fi
+
```
Comment (joeltimothyoh) [inline build.sh:105]: no reason for this section to be moved below

**Review by joeltimothyoh — CHANGES_REQUESTED** [review]

File: build.sh [hunk 3]
```diff
@@ -98,6 +92,17 @@ else
...
+# Process user variables
+REGISTRY_GOLDSOURCE=${REGISTRY_GOLDSOURCE:?err}
+REGISTRY_SOURCE=${REGISTRY_SOURCE:?err}
+if [ ! "$NO_PUSH" = 'true' ]; then
+    REGISTRY_USER=${REGISTRY_USER:?err}
+    REGISTRY_PASSWORD=${REGISTRY_PASSWORD:?err}
+else
+    REGISTRY_USER=${REGISTRY_USER:-}
+    REGISTRY_PASSWORD=${REGISTRY_PASSWORD:-}
+fi
```
Comment (joeltimothyoh) [inline build.sh:104]: no reason for this section to be moved below

**Review by joeltimothyoh — CHANGES_REQUESTED** [review]

File: build.sh (hunk 2 above)
Comment (leojonathanoh) [inline build.sh:105]: it needs `NO_PUSH` variable to exist

File: build.sh (hunk 3 above)
Comment (leojonathanoh) [reply build.sh:104]: it needs NO_PUSH variable to exist

Comment (joeltimothyoh) [reply build.sh:104]: sort them in the original order

**Pull request merged**